Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -10,5 +23,90 @@ namespace Unity.Multiplayer.Samples.BossRoom.Client
     public class ClientMainMenuState : GameStateBehaviour
     {
         public override GameState ActiveState { get { return GameState.MainMenu; } }
+
+        private GameNetPortal m_GameNetPortal;
+
+        private ClientGameNetPortal m_ClientNetPortal;
+
+
+        [SerializeField] private GameObject[] m_GameObjectsThatWillBeInjectedAutomatically;
+        private DIScope m_Scope;
+
+        [SerializeField] private NameGenerationData m_NameGenerationData;
+        [SerializeField] private LobbyUIMediator m_LobbyUIMediator;
+
+        [SerializeField] private CanvasGroup m_MainMenuButtons;
+        [SerializeField] private GameObject m_SignInSpinner;
+
+        private void Awake()
+        {
+            m_MainMenuButtons.interactable = false;
+            m_LobbyUIMediator.Hide();
+            DIScope.RootScope.InjectIn(this);
+        }
+
+        [Inject]
+        private void InjectDependenciesAndInitialize(AuthenticationServiceFacade authServiceFacade, LocalLobbyUser localUser, LocalLobby localLobby)
+        {
+            m_Scope = new DIScope(DIScope.RootScope);
+
+            m_Scope.BindInstanceAsSingle(m_NameGenerationData);
+            m_Scope.BindInstanceAsSingle(m_LobbyUIMediator);
+
+            var unityAuthenticationInitOptions = new InitializationOptions();
+
+#if UNITY_EDITOR
+            //The code below makes it possible for the clone instance to log in as a different user profile in Authentication service.
+            //This allows us to test services integration locally by utilising Parrelsync.
+            if (ClonesManager.IsClone())
+            {
+                Debug.Log("This is a clone project.");
+                var customArguments = ClonesManager.GetArgument().Split(',');
+
+                //second argument is our custom ID, but if it's not set we would just use some default.
+
+                var hardcodedProfileID = customArguments.Length > 1 ? customArguments[1] : "defaultCloneID";
+
+                unityAuthenticationInitOptions.SetProfile(hardcodedProfileID);
+            }
+#endif
+            authServiceFacade.DoSignInAsync(OnAuthSignIn,  OnSignInFailed, unityAuthenticationInitOptions);
```

Comment:
That means that it will no longer be possible to test it locally using multiple builds on the same machine though, right? Is there a way to circumvent this? Or should we no longer support this workflow?

Reply:
It's a good workflow question for the Authentication team and other services folks though...

Reply:
So what are the use cases where we want both a few players running locally and lobby (instead of direct IP connection).
Cause there are uses cases. If I want to debug my max limits or debug multiple players bumping into each other in the dungeon, or something and I don't want to organize a full playtest for this, using player builds is better than having 8 ParrelSync editors running...
For requiring us to be online, I guess we could want to test multiplayer players locally, connecting and updating a lobby, see how that fares? I wouldn't want to have to wait for the playtest to realize this isn't working.
Why would we want the profile switch to be automatic? Could we just add a dropdown in our debug menu to switch profile manually? (I'm unfamiliar with how the profile works, I'm be talking shit here)

Reply:
I've added a capability to process command line arguments, which allows to pass an arg to force the auth system to use a different profile. See PR description.